Clean up err codes (#67837)
rust-lang/rust: src/librustc_error_codes/error_codes/E0164.md

> Something which isn't a tuple struct nor a tuple variant was used as a pattern.

Reviewer: ```suggestion
Something which is neither a tuple struct nor a tuple variant was used as a pattern.
```

Erroneous code example:

```compile_fail,E0164
enum A {
    B,
    C,
}

impl A {
    fn new() {}
}

fn bar(foo: A) {
    match foo {
        A::new() => (), // error!
        _ => {}
    }
}
```

> This error means that an attempt was made to match something which isn't a tuple

Reviewer: ```suggestion
This error means that an attempt was made to match something which is neither
```

> struct nor a tuple variant. Only those two elements are allowed as pattern:

Reviewer: ```suggestion
a tuple struct nor a tuple variant. Only these two elements are allowed as pattern:
```
